src/net/sourceforge/kolmafia/persistence/MonsterDatabase.java

```diff
@@ -801,10 +812,11 @@ public static int fixMonsterAttributes(Map<Integer, Map<Attribute, Object>> upda
     int count = 0;
 
     String filename = "monsters.txt";
+    String outputFilename = "monsters.new.txt";
```

Comment:
should this method not be overwriting the original file?

Reply:
KoLmafia is willing to "override" built-in files with versions in the "data" directory.

If we use the same file name, the first time you run it, it reads from the built-in file from the .jar file and writes to "data". You can repeat the process as often as you want in the same session. Subesequent times it will find 0 monsters to change.

If you start a new session without rebuilding your jar, KoLmafia will find the newer file in "data" and use it as an override. If I run the script again, it reads the file from "data" and simultaneously writes the same file. I am unclear exactly why it works this way, but the end result is that rather than stopping at the natural EOF, it continues reading and writing an infinite number of blank lines until I abort the gCLI command.

I considered a couple of solutions:

- Use a "backup" system: if monsters.txt is in "data", rename it to monster.txt.bak (or something) before opening monsters.txt for output.
- Write to a different file name and copy that file to monsters.txt in src/data when it is time to commit.

Considering that even if I did the first approach, KoLmafia would not see it until you start a new session: it decides whether or not to use an "override" when you restart. And if you've rebuild the .jar, it will not use a file in "data".

The second approach seemed simpler. If you want to try out the new file, but it in src/data and rebuild.

If you have a different approach and feel strongly it is better, I'm happy to consider it.
Otherwise, I am content with how it is now.

Reply:
I can force it to always read from the built-in file and write to monsters.txt.
That means that if you restart KoLmafia without rebuilding, it will use the file in data as an override,  but it will still munge the built-in version - until you copy it into src/data.

Let me try that out.

Reply:
I am happy with that. After running the script, you can quit and restart and KoLmafia will use the file in data as an override and you can see how things worked out.